core/aws_ddk_core/resources/_kinesis.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Optional, Sequence
+
+import aws_cdk.aws_kinesisfirehose_alpha as firehose
+from aws_cdk.aws_iam import IRole
+from aws_cdk.aws_kinesis import IStream
+from aws_cdk.aws_kms import IKey
+from constructs import Construct
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class KinesisFactory:
```

Comment:
Tbh I'm not 100% sure we really need a factory here bc there's no configuration or defaults

Reply:
I can definitely imagine us having methods for Data Stream and potentially Kinesis Destinations in the near future.

Reply:
It seems that kinesis destinations would benefit more from configuration/defaults than a delivery stream. We might want to consider that

Reply:
Cool me too, happy to write that out